gammapy/utils/wcs.py

```diff
@@ -0,0 +1,133 @@
+    # In FITS, the edge of the image is at pixel coordinate +0.5.
+    refpix_x, refpix_y = w.wcs.crpix
+    refval_x, refval_y = w.wcs.crval
+    refval_x = Angle(refval_x, unit_x)
+    refval_y = Angle(refval_y, unit_y)
+    bin_edges_x += refval_x - (refpix_x - 0.5)*delta_x
+    bin_edges_y += refval_y - (refpix_y - 0.5)*delta_y
+
+    # set small values (compared to delta (i.e. step)) to 0
+    for i in np.arange(len(bin_edges_x)):
+        if np.abs(bin_edges_x[i]/delta_x) < 1.e-10:
+            bin_edges_x[i] = Angle(0., unit_x)
+    for i in np.arange(len(bin_edges_y)):
+        if np.abs(bin_edges_y[i]/delta_y) < 1.e-10:
+            bin_edges_y[i] = Angle(0., unit_y)
+
+    return bin_edges_x, bin_edges_y
+
+
+def linear_wcs_from_arrays(name_x, name_y, bin_edges_x, bin_edges_y):
+    """Make a 2D linear WCS object from arrays of bin edges.
+
+    This method gives the correct answer only for linear X, Y binning.
+    X is identified with WCS axis 1, Y is identified with WCS axis 2.
+
+    Parameters
+    ----------
+    name_x : `~string`
+        name of X coordinate, to be used as 'CTYPE' value
+    name_y : `~string`
+        name of Y coordinate, to be used as 'CTYPE' value
+    bin_edges_x : `~astropy.coordinates.Angle`
+        array with the bin edges for the X coordinate
+    bin_edges_y : `~astropy.coordinates.Angle`
+        array with the bin edges for the Y coordinate
+
+    Returns
+    -------
+    w : `~astropy.wcs.WCS`
+        WCS object describing the bin coordinates
+    """
+    # check units
+    unit_x = bin_edges_x.unit
+    unit_y = bin_edges_y.unit
+    if unit_x != unit_y:
+        ss_error = "Units of X ({0}) and Y ({1}) bins do not match!".format(
+            unit_x, unit_y)
+        ss_error += " Is this expected?"
+        raise ValueError(ss_error)
+
+    # check that bins are linear (edges equally spaced)
+    bin_widths_x = np.diff(bin_edges_x)
+    if not np.allclose(bin_widths_x, bin_widths_x[0]):
+        raise ValueError("The X bins are not linear! Diff = {}".format(np.diff(bin_edges_x.value)))
+    bin_widths_y = np.diff(bin_edges_y)
+    if not np.allclose(bin_widths_y, bin_widths_y[0]):
+        raise ValueError("The Y bins are not linear! Diff = {}".format(np.diff(bin_edges_y.value)))
+
+    # Create a new WCS object. The number of axes must be set from the start
+    w = WCS(naxis=2)
+
+    # Set up DET coordinates in degrees
+    nbins_x = len(bin_edges_x) - 1
+    nbins_y = len(bin_edges_y) - 1
+    range_x = Angle([bin_edges_x[0], bin_edges_x[-1]])
+    range_y = Angle([bin_edges_y[0], bin_edges_y[-1]])
+    delta_x = (range_x[1] - range_x[0])/nbins_x
+    delta_y = (range_y[1] - range_y[0])/nbins_y
+    w.wcs.ctype = [name_x, name_y]
+    w.wcs.cunit = [unit_x, unit_y]
+    w.wcs.cdelt = [delta_x.to(unit_x).value, delta_y.to(unit_y).value]
+    # ref as lower left corner (start of (X, Y) bin coordinates)
+    # coordinate start empiricaly determined at pix = 0.5: why 0.5?
```

Comment:
I think we discussed this and it's understood, so remove the "why 0.5?" comment?